### PR TITLE
[CDAP-15523] Adds initial set of e2e tests for DataPrep Connections.

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Fixes GCS, BigQuery & Spanner connections to have appropriate `data-cy` attributes to be able to refer from cypress
- Adds basic CRUD tests for GCS, BigQuery & Spanner DataPrep Connections

_**Miscellaneous:**_
- Fixes onEnter to delete connection in the delete confirmation modal
- Fixes anchor in GCP connections to be block level elements.
- Fixes minor typos in add and delete connections.
- Fixes the usage of `LoadingSVG` to be replaced by `LoadingSVGCentered` to avoid having the loading icon jump up and down.

**Note**
- This PR will initial fail the e2e tests in bamboo. I will work on setting up the bamboo environment to be able to run the tests. One that is done the tests should go back to green. In the mean time this can be tried locally. Please DM me for credentials.
- This PR does not add the actual test it needs (testing from pipeline studio) yet. Will add that once I fix the bamboo environment.

### CODE REVIEW

File: cdap-ui/cypress/support/commands.ts [hunk 1]
```diff
@@ -57,3 +59,99 @@ Cypress.Commands.add('cleanup_pipelines', (headers, pipelineName) => {
...
 });
+
+Cypress.Commands.add(
+  'fill_GCS_connection_create_form',
+  (
+    connectionId,
+    projectId = DEFAULT_GCP_PROJECTID,
+    serviceAccountPath = DEFAULT_GCP_SERVICEACCOUNT_PATH
+  ) => {
+    cy.visit('/cdap/ns/default/connections');
```
Comment (elfenheart) [inline cdap-ui/cypress/support/commands.ts:70]: set this path as a variable since it's being shared in a few places?

**Review by elfenheart — APPROVED** [review]
minor comment, otherwise 👍 LGTM

**Comment by ajainarayanan** [comment]
Update: This PR adds proper test cases and it runs successfully on various scenarios locally (provided the proper environment variables are set). Right now the bamboo build fails because of the requirement to place the service account file in the coopr cluster that runs CDAP. Need to figure out a way to make sure that is possible. Will update in a day or two 🤞

File: cdap-ui/cypress/support/commands.ts [hunk 2]
```diff
@@ -57,3 +59,212 @@ Cypress.Commands.add('cleanup_pipelines', (headers, pipelineName) => {
...
+
+Cypress.Commands.add(
+  'fill_BIGQUERY_connection_create_form',
+  (
+    connectionId,
+    projectId = DEFAULT_GCP_PROJECTID,
+    serviceAccountPath = DEFAULT_GCP_SERVICEACCOUNT_PATH
+  ) => {
+    cy.visit('/cdap/ns/default/connections');
+    cy.get('[data-cy="wrangler-add-connection-button"]', { timeout: 30000 }).click();
```
Comment (elfenheart) [inline cdap-ui/cypress/support/commands.ts:103]: it can take up to 30 seconds for adding a connection?

File: cdap-ui/cypress/support/commands.ts [hunk 3]
```diff
@@ -57,3 +59,212 @@ Cypress.Commands.add('cleanup_pipelines', (headers, pipelineName) => {
...
+      });
+      return cy.wrap(null);
+    }
+    let res = response.body;
+    if (res) {
+      if (typeof res === 'string') {
+        try {
+          res = JSON.parse(res);
+        } catch (e) {
+          cy.log('Got a invalid JSON response for wrangler service status: ', e, response.body);
```
Comment (elfenheart) [inline cdap-ui/cypress/support/commands.ts]: nit: `Got an invalid...`

**Review by elfenheart — APPROVED** [review]
minor comments, otherwise 👍 LGTM

File: cdap-ui/cypress/support/commands.ts (hunk 2 above)
Comment (ajainarayanan) [reply cdap-ui/cypress/support/commands.ts:103]: So this is to find the add-connection button on the bottom left of connections. UI could potentially take 30 seconds to show the connections page for the first time when we just deployed and enabled dataprep app. This is to make sure the loading indicator is gone and we are able to click on the add connection button.

**Comment by ajainarayanan** [comment]
Latest 3 consecutive green builds: [Build31](https://builds.cask.co/build/result/gotoBuildResult.action?buildKey=IT-UPD261&buildNumber=31&returnUrl=%2Fbrowse%2FIT-UPD261-31), [Build32](https://builds.cask.co/build/result/gotoBuildResult.action?buildKey=IT-UPD261&buildNumber=32&returnUrl=%2Fbrowse%2FIT-UPD261-31), [Build33](https://builds.cask.co/build/result/gotoBuildResult.action?buildKey=IT-UPD261&buildNumber=33&returnUrl=%2Fbrowse%2FIT-UPD261-31)

**Pull request merged**